iOS/Source/BookmarksController.swift

```diff
@@ -117,7 +140,10 @@ extension BookmarksController: UITableViewDelegate {
         }
 
         let editRowAction = UITableViewRowAction(style: .normal, title: "Edit") { (action, indexPath) in
-            // edit and update bookmark
+            guard let path = self.bookmarksDataSource.item(at: indexPath).path else { fatalError("!") }
+            let bookmarkController = BookmarkController(path: path, fetcher: self.fetcher, delegate: self)
```

Comment:
I see, is because you don't want the fetcher to know too much?

Fetcher works with objects, you send objects you get an error or not.

You ask for data, you either get a list of objects or a confirmation that it's all persisted.